src/main/scala/com/owlcyberdefense/dfdl/IPv4Checksum.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.owlcyberdefense.dfdl
+
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.layers.ByteBufferExplicitLengthLayerTransform
+import org.apache.daffodil.layers.LayerRuntimeInfo
+import org.apache.daffodil.layers.VariableHandle
+import passera.unsigned.UShort
+
+import java.nio.ByteBuffer
+
+/**
+ *  The layer transform computes the checksum of the header data.
+ *  per IETF RFC 791.
+ *
+ *  The data has a well-known fixed length, so layerLengthKind is always 'implicit'.
+ */
+final class IPv4Checksum(
+  name: String,
+  layerRuntimeInfo: LayerRuntimeInfo,
+  outputVar: VariableHandle)
+extends ByteBufferExplicitLengthLayerTransform[Int](
+  layerRuntimeInfo,
+  name,
+  inputVars = Seq(),
+  outputVar: VariableHandle) {
```

Comment:
Line 42 may compile, but it looks strange to me.  You're calling ByteBufferExplicitLengthLayerTransform's 4-parameter constructor with 4 arguments, but the 4th argument looks like a member field definition, not an argument being passed to ByteBufferExplicitLengthLayerTransform.  You should be able to remove the `: VariableHandle` type from line 42.

Also, Daffodil has a duplicate of this file too (with the same `: VariableHandle`): daffodil-test/src/test/scala/org/apache/daffodil/layers/IPv4Checksum.scala.

Reply:
I wasn't sure what to do about the fact that Daffodil has this same scala code built into it. The actual IPv4 example in daffodil-test is NOT exactly the same. It has fake IPv4 payload data, lacks all the ethernet frame surround, etc. 

But this particular scala file is identical. 

For at least test coverage reasons, this stuff needs to be part of daffodil also.

Reply:
The fact that this compiles with that type annotation at line 42 actually surprises me. But sure 'nuf.....